Ipu20000

PyMap opened the pull request:
Move `ignore_max_iters` boolean parameter from `household_weights` to `synthesize_all` method.

Main purpose of this change is to be able to decide if synthesis breaks or continues by exporting `npy` files to analyze block groups that didn't reach convergence.

Commit:
Skip max iterations

Commit:
Small fix

Wrong naming

PyMap committed:
Merge pull request #62 from UDST/add_dtype_serialno

Add dtype serialno

cvanoli committed:
Add ignore_max_iterations var to synthesize_all functions

cvanoli committed:
change return order for zone_synthesizer: first hh, then pp

Commit:
Add python 3.7 on setup

Commit:
Update drop_zeros for pandas compatibility

Commit:
max number of iterations parameter in household_weights method

Commit:
export npy for different geography types

Commit:
Include default ignore_max_iterarions parameters in zone synthesizer

Commit:
remove trace

Commit:
Resolve conflicts

Commit:
Update ipu tests

Commit:
Update ipu tests

Commit:
Update ipu tests

Commit:
update numpy verion

Commit:
pep8 styling

Commit:
Update geography parameter for household_weight test

Commit:
Remove python 2.7

Commit:
Restore python 2.7

Commit:
Rebase travis.yaml

https://github.com/UDST/synthpop/blob/991102b766fa5fa76c599eda98bc0af84b9a909c/.travis.yml

PyMap committed:
Merge branch 'develop' into IPU20000

msoltadeo commented on synthpop/ipu/ipu.py:
```diff
@@ -261,9 +263,33 @@ def household_weights(
+                if isinstance(geography, pd.Series):
+                    state, county = geography['state'], geography['county']
+                    tract, bgroup = geography['tract'], geography['block group']
+                    np.save('max_iter_{}_{}_{}_{}.npy'.format(state, county,
+                                                              tract, bgroup), ipu_dict)
+                elif isinstance(geography, list):
+                    np.save('max_iter_{}_{}.npy'.format(geography[0], geography[1]), ipu_dict)
+                else:
+                    np.save('max_iter_{}.npy'.format(str(geography)), ipu_dict)
```
@PyMap , can we add a year to the nane of the exported *npy file?

PyMap replied:
@msoltadeo , I've been doubting regarding this feature becasue I realized that if we add a `year` paramter to `synthesize` first and to `synthesize_all` secondly we will be changing the way these methods work just to give a name to a file in `household_weights`. And since it is something optional (users can decide to keep the default False for ignore_max_iters) we will need to add a new parameter when in some cases this won't be used. So, adapting all relative changes to this new parameter (unit tests, etc.) to a parameter that wont possibly be used is not convincing me. What do you think?

msoltadeo commented on synthpop/synthesizer.py:
```diff
@@ -25,7 +25,7 @@ def enable_logging():
 
 
-def synthesize(h_marg, p_marg, h_jd, p_jd, h_pums, p_pums,
+def synthesize(h_marg, p_marg, h_jd, p_jd, h_pums, p_pums, geography, ignore_max_iters,
```
@PyMap we have different default ignore_max_iters values for different functions. If we are setting a default year, I suggest setting `False` since that is how the synthpop is working now.

PyMap replied:
@msoltadeo you are right. I set up the `ignore_max_ters` as True to check the unit tests. I think that by changing the default in `synthesize_all` from True to False we will achieve what you are pointing out (keep the way synthpop used to work)

msoltadeo reviewed (commented):
@PyMap just a couple of small suggestiongs

Commit:
Restore ignore_max_iters default

I set it up as True to check unit tests. Now that it works, I'm restoring to to the previous usage
